ui/components/app/collectibles-items/collectibles-items.js

```diff
@@ -0,0 +1,202 @@
+                      const collectibleImage = getAssetImageURL(
+                        image,
+                        ipfsGateway,
+                      );
+                      return (
+                        <Box width={width} margin={1} key={`collectible-${i}`}>
+                          <div
+                            className="collectibles-items__image__wrapper"
+                            style={{
+                              backgroundColor,
+                            }}
+                          >
+                            <img
+                              onClick={() =>
+                                history.push(
+                                  `${ASSET_ROUTE}/${address}/${tokenId}`,
+                                )
+                              }
+                              className="collectibles-items__image"
+                              src={collectibleImage}
+                            />
+                          </div>
+                        </Box>
+                      );
+                    })}
+                  </Box>
+                ) : null}
+              </div>
+            );
+          })}
+          <Box
+            marginTop={6}
+            flexDirection={FLEX_DIRECTION.COLUMN}
+            justifyContent={JUSTIFY_CONTENT.CENTER}
+          >
+            <Typography
+              color={COLORS.UI3}
+              variant={TYPOGRAPHY.H5}
+              align={TEXT_ALIGN.CENTER}
+            >
+              {t('missingNFT')}
+            </Typography>
+            <Box
+              alignItems={ALIGN_ITEMS.CENTER}
+              justifyContent={JUSTIFY_CONTENT.CENTER}
+            >
+              <Box
+                className="collectibles-items__link"
+                justifyContent={JUSTIFY_CONTENT.FLEX_END}
+              >
+                {useCollectibleDetection ? (
+                  <Button type="link" onClick={onRefreshList}>
+                    {t('refreshList')}
+                  </Button>
+                ) : (
+                  <Button type="link" onClick={onEnableAutoDetect}>
+                    {t('enableAutoDetect')}
+                  </Button>
+                )}
+              </Box>
+              <Typography
+                color={COLORS.UI3}
+                variant={TYPOGRAPHY.H4}
+                align={TEXT_ALIGN.CENTER}
+              >
+                {t('or')}
+              </Typography>
+              <Box
+                justifyContent={JUSTIFY_CONTENT.FLEX_START}
+                className="collectibles-items__link"
+              >
+                <Button type="link" onClick={onAddNFT}>
+                  {t('importNFTs')}
+                </Button>
+              </Box>
+            </Box>
+          </Box>
+        </>
+      </Box>
+    </div>
+  );
+}
+
+CollectiblesItems.propTypes = {
+  onAddNFT: PropTypes.func.isRequired,
+  onRefreshList: PropTypes.func.isRequired,
+  collections: PropTypes.array,
```

Comment:
Is there a way to specify the shape of this array? Should this be required too?

Reply:
added improved typing here: https://github.com/MetaMask/metamask-extension/pull/12970/commits/5ec71ea225d74795b08c5206f38e420a4729d6a0